addons/pos_gift_card/static/src/js/GiftCardPopup.js

```diff
@@ -126,12 +126,12 @@ odoo.define("pos_gift_card.GiftCardPopup", function (require) {
           this.env.pos.config.gift_card_product_id[0]
         ];
 
-      for (let line of order.orderlines.models) {
-        if (line.product.id === giftProduct.id && line.price < 0) {
-          if (line.gift_card_id === (await this.getGiftCard()).id) return line;
-        }
-      }
-      return false;
+      const gitfCard = await this.getGiftCard();
+      return order.get_orderlines().filter(
+          line => line.get_product().id === giftProduct.id
+          && line.price < 0
```

Comment:
Why do we need to evaluate `price < 0`?

Exists some case where a giftcard will be added and will have a positive amount?

If the later happens, why would you exclude that line? If it is a giftcard?

Reply:
I'm just using the `ifs` in the original code and I pass it as a `filter`, that validation was part of the `ifs`